src/runtime/extern.go

```diff
@@ -3,3 +3,10 @@ package runtime
 func Callers(skip int, pc []uintptr) int {
 	return 0
 }
+
+// Version returns the Go tree's version string.
+// It is either the commit hash and date at the time of the build or,
+// when possible, a release tag like "go1.3".
+func Version() string {
+	return "TODO: not implemented"
```

Comment:
This is actually something that's not too difficult to support. The compiler already supports the `-X` flag and we could add the version relatively easily. See these lines for a starting point:

https://github.com/tinygo-org/tinygo/blob/3883550c446fc8b14115366284316290f3d3c4b1/builder/build.go#L846-L850

We could for example add `runtime.buildVersion` to `GlobalValues` earlier in the build. The version itself can be obtained using `goenv.GorootVersionString`.

Reply:
I've updated the PR to your suggestion.
Now the tinygo runtime.Version() is always the version of golang that was used to build tinygo, but not the version of tinygo itself, which I think would be more useful?

Also, for some reason I wasn't able to set the version via `go build -tags=llvm11 -ldflags="-X 'runtime.buildVersion=abc'" -o build/tinygo` - shouldn't that work too, so that tinygo could set its own version during build?

Reply:
> Now the tinygo runtime.Version() is always the version of golang that was used to build tinygo, but not the version of tinygo itself, which I think would be more useful?

I agree that we should return the version of TinyGo used to build the binary here, which is what the main Go seems to do with respect to the version of Go used to build binary.

Reply:
I've moved and exported `main.gitSha1` to `goenv.GitSha1` to be able to access it from both the `main` and `builder` package.
Now the `runtime.Version()` function returns a version string similar to what `tinygo version` returns (`0.23.0` - or, if it ends in `-dev` it will be the concatenation of the version and git sha `0.23.0-dev-f5170c0b`).

https://github.com/tinygo-org/tinygo/pull/2676/files#diff-d871f892c3b7d415467578ac1742cccab676ec4cc21022b0b9b34f5741237eb7R208-R217